src/main/java/io/cryostat/recordings/RecordingArchiveHelper.java

```diff
@@ -255,19 +288,38 @@ private void validateSavePath(String recordingName, Path path) throws IOExceptio
         }
     }
 
-    public boolean deleteReport(String recordingName) {
+    public boolean deleteReport(String sourceTarget, String recordingName) {
         try {
             logger.trace("Invalidating archived report cache for {}", recordingName);
-            return fs.deleteIfExists(getCachedReportPath(recordingName));
-        } catch (IOException ioe) {
-            logger.warn(ioe);
+            return fs.deleteIfExists(getCachedReportPath(sourceTarget, recordingName).get());
+        } catch (IOException | InterruptedException | ExecutionException e) {
+            logger.warn(e);
             return false;
         }
     }
 
-    public Path getCachedReportPath(String recordingName) {
+    public Future<Path> getCachedReportPath(String sourceTarget, String recordingName) {
+        CompletableFuture<Path> future = new CompletableFuture<>();
+
+        String subdirectory;
+        if (sourceTarget == null) {
+            subdirectory = "default";
+        } else if (sourceTarget.equals(UPLOADED_RECORDINGS_SUBDIRECTORY)) {
+            subdirectory = UPLOADED_RECORDINGS_SUBDIRECTORY;
+        } else {
+            subdirectory = base32.encodeAsString(sourceTarget.getBytes(StandardCharsets.UTF_8));
+        }
         String fileName = recordingName + ".report.html";
-        return archivedRecordingsReportPath.resolve(fileName).toAbsolutePath();
+
+        try {
+            Path tempSubdirectory =
+                    fs.createTempDirectory(archivedRecordingsReportPath, subdirectory);
```

Comment:
I didn't have time to familiarize myself with the `java.nio.file.Files` implementation of temporary directories so this behavior is likely incorrect. Reports are generated correctly but I think this results in a new subdirectory being created each time this method is called, even if one of the same `subdirectory` name already exists. My understanding is that the `String prefix` is used to generate candidate names for the directory which makes it hard to check for the existence of a subdirectory before creating it, unlike with regular, non-temp directories.  However, I could be wrong. Requires further investigation.

Reply:
This sounds correct to me.

Reply:
With this behaviour, I think we could use  `archivedRecordingsReportPath.resolve(subdirectory)` as cached report directory so we can use `exists` to check if needed for `createDirectory`?

Reply:
Or maybe there is a specific reason to enforce `createTempDirectory`?

Reply:
Sorry, I meant that the "my understanding..." part was correct, not necessarily that the use of a temp directory is correct. I haven't really looked at this yet.